{{cookiecutter.project_name}}/justfile

```diff
@@ -0,0 +1,196 @@
+# On Windows the bash shell that comes with Git for Windows should be used.
+# If it is not on path, give the path to the executable in the following line.
+#set windows-shell := ["C:/Program Files/Git/usr/bin/sh", "-cu"]
+
+# Load environment variables from config.public.mk or specified file
+set dotenv-load := true
+# set dotenv-filename := env_var_or_default("LINKML_ENVIRONMENT_FILENAME", "config.public.mk")
+set dotenv-filename := x'${LINKML_ENVIRONMENT_FILENAME:-config.public.mk}'
+
+
+# List all commands as default command. The prefix "_" hides the command.
+_default: _status
+    @just --list
+
+# Set cross-platform Python shebang line (assumes presence of launcher on Windows)
+shebang := if os() == 'windows' {
+  'py'
+} else {
+  '/usr/bin/env python3'
+}
+
+# Environment variables with defaults
+schema_name := env_var_or_default("LINKML_SCHEMA_NAME", "")
+source_schema_path := env_var_or_default("LINKML_SCHEMA_SOURCE_PATH", "")
+
+sheet_module := env_var_or_default("LINKML_SCHEMA_GOOGLE_SHEET_MODULE", "")
+sheet_ID := env_var_or_default("LINKML_SCHEMA_GOOGLE_SHEET_ID", "")
+sheet_tabs := env_var_or_default("LINKML_SCHEMA_GOOGLE_SHEET_TABS", "")
+sheet_module_path := source_schema_path / sheet_module + ".yaml"
+
+config_yaml := if env_var_or_default("LINKML_GENERATORS_CONFIG_YAML", "") == "" {
+  "--config-file " + env_var_or_default("LINKML_GENERATORS_CONFIG_YAML", "")
+} else {
+  ""
+}
+gen_doc_args := env_var_or_default("LINKML_GENERATORS_DOC_ARGS", "")
+gen_owl_args := env_var_or_default("LINKML_GENERATORS_OWL_ARGS", "")
+gen_java_args := env_var_or_default("LINKML_GENERATORS_JAVA_ARGS", "")
+gen_ts_args := env_var_or_default("LINKML_GENERATORS_TYPESCRIPT_ARGS", "")
+
+# Directory variables
+src := "src"
+dest := "project"
+pymodel := src / schema_name / "datamodel"
+docdir := "docs"
+exampledir := "examples"
+
+# Show current project status
+_status: _check-config
+    @echo "Project: {{schema_name}}"
+    @echo "Source: {{source_schema_path}}"
+
+# Run initial setup (run this first)
+setup: _check-config _git-init install _gen-project _gen-examples _gendoc _git-add _git-commit
+
+# Install project dependencies
+install:
+    poetry install
+
+# Check project configuration
+_check-config:
+    #!{{shebang}}
+    import os
+    schema_name = os.getenv('LINKML_SCHEMA_NAME')
+    if not schema_name:
+        print('**Project not configured**:\n - See \'.env.public\'')
+        exit(1)
+    print('Project-status: Ok')
+
+# Updates project template and LinkML package
+update: _update-template _update-linkml
+
+# Update project template
+_update-template:
+    cruft update
+
+# Update LinkML to latest version
+_update-linkml:
+    poetry add -D linkml@latest
+
+# Create data harmonizer
+_create-data-harmonizer:
+    npm init data-harmonizer {{source_schema_path}}
+
+# Generate all project files
+alias all := site
+
+# Generate site locally
+site: _gen-project _gendoc
+
+# Deploy site
+deploy: site
```

Comment:
Hmm, these do look like recreations of the same things in our Makefile. Or am I misunderstanding? Does `just` play nice with `make` or do we have two command runners with their own configs?

Reply:
Two runners, one config.